components-core/src/main/java/com/adyen/checkout/components/core/internal/analytics/DefaultAnalyticsManager.kt

```diff
@@ -101,31 +105,23 @@ internal class DefaultAnalyticsManager(
     }
 
     private suspend fun sendEvents() {
-        val checkoutAttemptId = checkoutAttemptId
-        if (checkoutAttemptId == null) {
-            adyenLog(AdyenLogLevel.WARN) { "checkoutAttemptId should not be null at this point." }
-            return
-        }
-
-        if (cannotSendEvents()) {
-            adyenLog(AdyenLogLevel.DEBUG) { "Not allowed to send events, ignoring." }
+        val checkoutAttemptIdState = checkoutAttemptIdState as? CheckoutAttemptIdState.Available
+        if (checkoutAttemptIdState == null) {
+            adyenLog(AdyenLogLevel.WARN) { "checkoutAttemptId should be available at this point." }
             return
         }
 
         runSuspendCatching {
-            analyticsRepository.sendEvents(checkoutAttemptId)
+            analyticsRepository.sendEvents(checkoutAttemptIdState.checkoutAttemptId)
         }.fold(
             onSuccess = { /* Not necessary */ },
             onFailure = { throwable -> adyenLog(AdyenLogLevel.WARN, throwable) { "Failed sending analytics events" } },
         )
     }
 
-    override fun getCheckoutAttemptId(): String? = checkoutAttemptId
+    override fun getCheckoutAttemptIdState() = checkoutAttemptIdState
```

Comment:
I think the state should be something private to `DefaultAnalyticsManager` and then this method can stay as it was

Reply:
That was my initial implementation, but I changed it, since `checkoutAttemptId-not-fetched` and `fetch-checkoutAttemptId-failed` values are not really related to the analytics. They are only necessary to be sent in the `/payments` call, based on if the `checkoutAttemptId` is fetched or not. Those two values are not the actual values of the `checkoutAttemptId`, that is why I separated them, to make sure that the Analytics manager is not aware of them and if they change in the future, they will not change the logic of the analytics manager. They are relevant only for the `/payments` call.

Do you think there is a better way to handle this and keep them separate from the analytics manager?

Reply:
Hmm I see your point, but I think it actually is related to analytics. Currently, the `getCheckoutAttemptId` method is only used by delegates for the `/payments` call. The only other place we use it is to make analytics requests, but we don't make analytics requests if we have no attempt id. Making the not fetched and failed values valid attempt ids imo

Reply:
I was thinking of exposing a state and whenever necessary we can assign custom values to the `checkoutAttemptId`, since those are not values which are returned from backend. But I guess it is just thinking about cases which we do not have now.
I can move everything to the analytics manager and later we can make changes if necessary.

Reply:
Yes, I think that's a good idea